siuba/experimental/pivot/__init__.py

```diff
@@ -16,6 +17,45 @@ def pivot_longer(
         names_repair = "check_unique",
         values_to = "value",
         values_drop_na = False,
-        values_ptypes = tuple()
+        values_ptypes = tuple(),
+        values_transform = dict(),
         ):
-    return gather(__data, names_to, values_to, cols)
+    
+    # Copied selection over from gather, maybe this can be compartmentalised away?
```

Comment:
I'd be interested on your thoughts re this comment (and the copied lines).

When I move on to `pivot_wider` I'll be using the same selection mechanism again, unless a different way makes more sense?

Reply:
This makes complete sense! (my guess is that at some point, `gather` could be re-written as a special case of `pivot_wider`, since it uses `unstack` under the hood.)

Reply:
Good call on the rewrite. On the `tidyr` front, they still maintain entirely separated functions for `gather`/`spread` and `pivot_longer`/`pivot_wider`. I'm guessing it's just legacy based? Not sure why they didn't just integrate the two.

Reply:
I wonder if it's related to things like pivot_longer outputting in a slightly different order than gather? I'm still trying to wrap my head around all the different things the pivot funcs can do..!